### PR TITLE
Add support for `disksup:get_disk_info/[0,1]`

### PR DESCRIPTION
**Pull request by lukebakken**
References #6156

https://github.com/erlang/otp/issues/6156#issuecomment-1208010552

This PR adds support for a function that will immediately fetch total and available disk space for all local drives, or for a specific path.

### CODE REVIEW

File: lib/os_mon/doc/src/disksup.xml [hunk 1]
```diff
@@ -111,6 +111,46 @@
...
+        <v>&nbsp;AvailableKiB = int()</v>
+        <v>&nbsp;Capacity = int()</v>
+      </type>
+      <desc>
+        <p>Immediately fetches total space, available space and capacity for local disks.
+          <c>Id</c> is a string that identifies the disk or partition.
+          <c>TotalKiB</c> is the total size of the disk or partition in kbytes.
+          <c>AvailableKiB</c> is the disk space used in kbytes.
+          <c>Capacity</c> is the percentage of disk space used.</p>
+        <p>Returns <c>[{"none",0,0,0}]</c> if <c>disksup</c> is not
```
Comment (AndrewDryga) [inline lib/os_mon/doc/src/disksup.xml:130]: Would it be better if we return an empty list or an error in this case instead?

Comment (lukebakken) [reply lib/os_mon/doc/src/disksup.xml:130]: 🤷 that's how `disksup` works now. My goal is to not change the existing API.

**Comment by lukebakken** [comment]
All set, cc @garazdawi

**Review by rickard-green — CHANGES_REQUESTED** [review]
Looks overall good to me.

File: lib/os_mon/doc/src/disksup.xml [hunk 2]
```diff
@@ -111,6 +111,47 @@
           available.</p>
       </desc>
     </func>
+    <func>
+      <name since="">get_disk_info() -> [DiskData]</name>
```
[the remark this answers is not in the document]
Comment (lukebakken) [reply lib/os_mon/doc/src/disksup.xml:115]: Thanks! Is there any way I could have looked up `OTP-18303`?

**Review by rickard-green — APPROVED** [review]
I removed the following from the commit message and force pushed:
```
Implement get_disk_info

Add failing test for disksup:get_disk_info/1

Implement `disksup:get_disk_info/1`

Documentation for `disksup:get_disk_info/1`

Handle case where `Path` does not return information

The command `df /` might return a list of info

Use os_mon_sysinfo:get_disk_info/0 for all drives on win32
```

We'll have a technical board meeting on Thursday where this PR will be brought up. I expect it to be accepted.

**Comment by rickard-green** [comment]
This was approved, but it was pointed out that the documentation used the term *kbytes* instead of *kibibytes*. I've changed that and force pushed the changes. Once the doc build has completed successfully I'll merge this.

**Pull request merged**